apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SessionPoolManager.cs

```diff
@@ -161,12 +162,20 @@ internal void Release(SessionPool sessionPool)
         internal IReadOnlyList<Statistics> GetStatistics() =>
             _targetedPools.ToArray().Select(tp => tp.Value.GetStatisticsSnapshot()).ToList().AsReadOnly();
 
-        internal SessionPool.DatabaseStatistics GetDatabaseStatistics(SpannerClientCreationOptions options, DatabaseName databaseName)
+        internal SessionPoolKeyStatistics GetDatabaseStatistics(SpannerClientCreationOptions options, SessionPoolKey key)
         {
             GaxPreconditions.CheckNotNull(options, nameof(options));
-            GaxPreconditions.CheckNotNull(databaseName, nameof(databaseName));
             _targetedPools.TryGetValue(options, out var targetedPool);
-            return targetedPool?.SessionPoolOrNull?.GetStatisticsSnapshot(databaseName);
+            return targetedPool?.SessionPoolOrNull?.GetStatisticsSnapshot(GaxPreconditions.CheckNotNull(key, nameof(key)));
```

Comment:
Consider moving the key null validation to its own line as it was before for the database name, here and everywhere, it's hardish to read now.